tests/basic_test.go

```diff
@@ -77,3 +81,123 @@ func TestJobSubmission(t *testing.T) {
 		t.Error(err.Error())
 	}
 }
+
+func TestSparkHistoryServerInstallation(t *testing.T) {
+	awsAccessKey := utils.GetenvOr("AWS_ACCESS_KEY_ID", "")
+	awsAccessSecret := utils.GetenvOr("AWS_SECRET_ACCESS_KEY", "")
+	awsSessionToken := utils.GetenvOr("AWS_SESSION_TOKEN", "")
+	awsBucketName := utils.GetenvOr("AWS_BUCKET_NAME", "infinity-artifacts-ci")
+	awsFolderPath := "/autodelete7d/spark-operator-history-server-test/"
+
+	// Make sure folder is deleted
+	err := utils.AwsS3DeleteFolder(awsBucketName, awsFolderPath)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+	// Make sure folder is created
+	err = utils.AwsS3CreateFolder(awsBucketName, awsFolderPath)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	awsBucketPath := "s3a://" + awsBucketName + awsFolderPath
+
+	historyParams := make(map[string]string)
+	historyParams["enableHistoryServer"] = "true"
+	historyParams["historyServerFsLogDirectory"] = awsBucketPath
+
+	if len(awsSessionToken) > 0 {
+		historyParams["historyServerOpts"] = "-Dspark.hadoop.fs.s3a.access.key=" + awsAccessKey +
+			" -Dspark.hadoop.fs.s3a.secret.key=" + awsAccessSecret +
+			" -Dspark.hadoop.fs.s3a.session.token=" + awsSessionToken +
+			" -Dspark.hadoop.fs.s3a.aws.credentials.provider=org.apache.hadoop.fs.s3a.TemporaryAWSCredentialsProvider" +
+			" -Dspark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem"
+	} else {
+		historyParams["historyServerOpts"] = "-Dspark.hadoop.fs.s3a.access.key=" + awsAccessKey +
+			" -Dspark.hadoop.fs.s3a.secret.key=" + awsAccessSecret +
+			" -Dspark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem"
+	}
```

Comment:
may I suggest refactoring it to avoid duplication? something like that:
```
historyParams["historyServerOpts"] = "-Dspark.hadoop.fs.s3a.access.key=" + awsAccessKey +
			" -Dspark.hadoop.fs.s3a.secret.key=" + awsAccessSecret +
			" -Dspark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem"

if len(awsSessionToken) > 0 {
		historyParams["historyServerOpts"] = historyParams["historyServerOpts"] + 
		" -Dspark.hadoop.fs.s3a.session.token=" + awsSessionToken +
		" -Dspark.hadoop.fs.s3a.aws.credentials.provider=org.apache.hadoop.fs.s3a.TemporaryAWSCredentialsProvider"
}
```